src/openms_gui/source/VISUAL/TOPPASScene.cpp

```diff
@@ -480,26 +488,64 @@ namespace OpenMS
 
   bool TOPPASScene::isEdgeAllowed_(TOPPASVertex* u, TOPPASVertex* v)
   {
-    if (u == 0 ||
-        v == 0 ||
-        u == v ||
+    if (u == 0 || v == 0 || u == v ||
         // edges leading to input files make no sense:
         qobject_cast<TOPPASInputFileListVertex*>(v) ||
         // neither do edges coming from output files:
         qobject_cast<TOPPASOutputFileListVertex*>(u) ||
-        // nor edges from input to output without a tool in between:
-        (qobject_cast<TOPPASInputFileListVertex*>(u)
-        && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
-        // nor multiple incoming edges for a single output file/list node
-        (qobject_cast<TOPPASOutputFileListVertex*>(v)
-        && v->inEdgesBegin() != v->inEdgesEnd()) ||
-        // nor mergers connected directly to an output node
-        (qobject_cast<TOPPASMergerVertex*>(u)
-        && qobject_cast<TOPPASOutputFileListVertex*>(v)))
+        // nor edges from input/merger/splitter directly to output:
+        ((qobject_cast<TOPPASInputFileListVertex*>(u) || qobject_cast<TOPPASMergerVertex*>(u) || qobject_cast<TOPPASSplitterVertex*>(u)) && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
+        // nor multiple incoming edges for an output or splitter node:
```

Comment:
Ok, I see your point (although I'm not entirely sure what you mean by 
multiple ways of merging, I think there can only be one: make n output 
rounds with 1 file each from a+b+c+..=n input files). But anyhow, lets 
leave it as it is :)


Am 9/15/2015 um 11:29 AM schrieb hendrikweisser:
> In src/openms_gui/source/VISUAL/TOPPASScene.cpp
> <https://github.com/OpenMS/OpenMS/pull/1593#discussion_r39490613>:
>
>>          // edges leading to input files make no sense:
>>          qobject_cast<TOPPASInputFileListVertex*>(v) ||
>>          // neither do edges coming from output files:
>>          qobject_cast<TOPPASOutputFileListVertex*>(u) ||
>> -        // nor edges from input to output without a tool in between:
>> -        (qobject_cast<TOPPASInputFileListVertex*>(u)
>> -        && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
>> -        // nor multiple incoming edges for a single output file/list node
>> -        (qobject_cast<TOPPASOutputFileListVertex*>(v)
>> -        && v->inEdgesBegin() != v->inEdgesEnd()) ||
>> -        // nor mergers connected directly to an output node
>> -        (qobject_cast<TOPPASMergerVertex*>(u)
>> -        && qobject_cast<TOPPASOutputFileListVertex*>(v)))
>> +        // nor edges from input/merger/splitter directly to output:
>> +        ((qobject_cast<TOPPASInputFileListVertex*>(u) || qobject_cast<TOPPASMergerVertex*>(u) || qobject_cast<TOPPASSplitterVertex*>(u)) && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
>> +        // nor multiple incoming edges for an output or splitter node:
>
> Thanks for the support, Lars, but Collectors can actually have multiple
> input edges. I considered removing this ability to make the behaviour
> more straight-forward, but didn't because that might have impacted
> existing workflows.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMS/OpenMS/pull/1593/files#r39490613>.
>